test/tests/problems/reference_residual_problem/tests

```diff
@@ -61,10 +61,11 @@
                   'of those variables.'
   []
   [abs_ref_acceptable]
-    type = 'CSVDiff'
+    type = 'RunApp'
     input = 'abs_ref_acceptable.i'
-    csvdiff = 'abs_ref_acceptable_out.csv'
-    rel_err = 1e-3
+    expect_out = 'Converged due to ACCEPTABLE tolerances'
+    max_threads = 1
+    max_parallel = 1
```

Comment:
This limitation is necessary? You don't get convergence with multiple threads or processes? It seems like we'd really like to be able to be confident that we can fulfill the requirement outlined in this test for multithreaded and multiprocess runs

Reply:
Yeah, it's probably okay, but was just trying to limit based on suggestions by @permcody . I can probably get the tests passing now, but worried about future proofing.

Reply:
Being able to pass in many different configurations is one of the best ways to future proof. This is akin to why we now test everything with different PETSc versions. This may pass in serial now, but a PETSc update (for example), may slightly change the solve and then it will no longer pass in serial. However, if the test passes in serial, with mpi, with threads, etc. then it lends more confidence that the test will be stable across different future architectures, mpi implementations, future petsc versions, etc.

Reply:
Sure, let me remove these then.

Reply:
Being able to pass in a lot of different configurations suggests the test is well conditioned and more robust against perturbations